app/controllers/discourse_automation/append_last_checked_by_controller.rb

```diff
@@ -0,0 +1,24 @@
+# frozen_string_literal: true
+
+module DiscourseAutomation
+  class AppendLastCheckedByController < ApplicationController
+    requires_login
+
+    def post_checked
+      post = Post.find(params[:post_id])
+      guardian = Guardian.new(current_user)
+      raise Discourse::InvalidAccess.new if !guardian.can_edit?(post)
```

Comment:
This can be shorted by calling the `ensure_<action>!` method on the guardian 😉 

```suggestion
      guardian.ensure_can_edit!(post)
```